health/polling.go

```diff
@@ -157,35 +157,38 @@ func (r *pollingChecker) New(
 		defer ticker.Stop()
 
 		for {
-			ctx, cancel := context.WithTimeout(ctx, r.timeout)
-			defer cancel()
-
-			result := r.prober.Probe(ctx, conn)
-
-			lastState := state
-			switch {
-			case result == StateHealthy && (state == StateUnhealthy || state == StateDegraded || state == StateUnknown):
-				counter++
-				if counter >= r.healthyThreshold {
+			func() {
+				ctx, cancel := context.WithTimeout(ctx, r.timeout)
+				defer cancel()
+
+				result := r.prober.Probe(ctx, conn)
```

Comment:
I didn't change this, which means the timeout is implicit, relying on `Probe` to use the context's timeout (we see such a change in the unit test). To handle timeout here, I think we would need to spawn a goroutine here for the probe - wasn't sure if that's overkill or not. 

Personally I would move timeout parameter off of the poller and onto the prober's themselves, though it does mean a user has to set both interval and timeout meaningfully themselves so maybe not